client/src/app/core/ui-services/base-import.service.ts

```diff
@@ -286,11 +287,10 @@ export abstract class BaseImportService<V extends BaseViewModel> {
      * @param event type is Event, but has target.files, which typescript doesn't seem to recognize
      */
     public onSelectFile(event: any): void {
-        // TODO type
         if (event.target.files && event.target.files.length === 1) {
-            if (event.target.files[0].type.startsWith('text/csv')) {
+            if ((event.target.files[0] as File).type.startsWith('text/csv')) {
                 this._rawFile = event.target.files[0];
-                this.readFile(event.target.files[0]);
+                this.readFile(event.target.files[0] as File);
             } else {
```

Comment:
Still not the best solution. Typescript doesn't currently complain and it should be clear what is arriving, though